bin/packages/build.js

```diff
@@ -1,220 +1,82 @@
-			.then( ( result ) => callback( null, result ) );
-	};
-
-	const postCSSRTLSync = ( ltrCSS, callback ) => {
-		postcss( [ require( 'rtlcss' )() ] )
-			.process( ltrCSS, { from: 'src/app.css', to: 'dest/app.css' } )
-			.then( ( result ) => callback( null, result ) );
-	};
-
-	const result = deasync( postCSSSync )();
-	fs.writeFileSync( outputFile, result.css );
-
-	const resultRTL = deasync( postCSSRTLSync )( result );
-	fs.writeFileSync( outputFileRTL, resultRTL );
-}
-
-/**
- * Build a file for a specific environment
- *
- * @param {string}  file        File path to build
- * @param {boolean} silent      Show logs
- * @param {string}  environment Dist environment (node or es5)
- */
-function buildJsFileFor( file, silent, environment ) {
-	const buildDir = BUILD_DIR[ environment ];
-	const destPath = getBuildPath( file, buildDir );
-	const babelOptions = getBabelConfig( environment, file.replace( PACKAGES_DIR, '@wordpress' ) );
+	bar.tick( 0 );
 
-	mkdirp.sync( path.dirname( destPath ) );
-	const transformed = babel.transformFileSync( file, babelOptions );
-	fs.writeFileSync( destPath + '.map', JSON.stringify( transformed.map ) );
-	fs.writeFileSync( destPath, transformed.code + '\n//# sourceMappingURL=' + path.basename( destPath ) + '.map' );
-
-	if ( ! silent ) {
-		process.stdout.write(
-			chalk.green( '  \u2022 ' ) +
-				path.relative( PACKAGES_DIR, file ) +
-				chalk.green( ' \u21D2 ' ) +
-				path.relative( PACKAGES_DIR, destPath ) +
-				'\n'
-		);
-	}
-}
-
-/**
- * Build the provided package path
- *
- * @param {string} packagePath absolute package path
- */
-function buildPackage( packagePath ) {
-	const srcDir = path.resolve( packagePath, SRC_DIR );
-	const jsFiles = glob.sync( `${ srcDir }/**/*.js`, {
+	stream = glob.stream( [
+		`${ PACKAGES_DIR }/*/src/**/*.js`,
+		`${ PACKAGES_DIR }/*/src/*.scss`,
+	], {
 		ignore: [
-			`${ srcDir }/**/test/**/*.js`,
-			`${ srcDir }/**/__mocks__/**/*.js`,
+			`**/test/**`,
+			`**/__mocks__/**`,
 		],
-		nodir: true,
+		onlyFiles: true,
 	} );
 
-	process.stdout.write( `${ path.basename( packagePath ) }\n` );
+	// Pause to avoid data flow which would begin on the `data` event binding,
+	// but should wait until worker processing below.
+	//
+	// See: https://nodejs.org/api/stream.html#stream_two_reading_modes
+	stream
+		.pause()
+		.on( 'data', ( file ) => {
+			bar.total = files.push( file );
+		} );
+
+	onFileComplete = () => {
+		bar.tick();
+	};
+}
 
-	// Build js files individually.
-	jsFiles.forEach( ( file ) => buildJsFile( file, true ) );
+const worker = workerFarm( require.resolve( './build-worker' ) );
 
-	// Build package CSS files
-	buildPackageScss( packagePath );
+let ended = false,
+	complete = 0;
 
-	process.stdout.write( `${ DONE }\n` );
-}
+stream
+	.on( 'data', ( file ) => worker( file, ( error ) => {
+		onFileComplete();
 
-const files = process.argv.slice( 2 );
+		if ( error ) {
+			console.log( error );
```

Comment:
I think it should be `console.error`, so that the output is printed to `stderr`.
```suggestion
			console.error( error );
```

Reply:
> I think it should be `console.error`, so that the output is printed to `stderr`.

Makes sense! Updated in d797c5b.